hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/file/OMDirectoryCreateRequest.java

```diff
@@ -139,12 +140,15 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
       acquiredLock = omMetadataManager.getLock().acquireLock(BUCKET_LOCK,
           volumeName, bucketName);
 
-      // TODO: Not checking volume exist here, once we have full cache we can
-      //  add volume exist check also.
+      // Check volume exist.
+      String volumeKey = omMetadataManager.getVolumeKey(volumeName);
+      if (!omMetadataManager.getVolumeTable().isExist(volumeKey)) {
+        throw new OMException("Volume not found " + volumeName,
```

Comment:
This can be moved to inside if (omBucketInfo == null) to Line 153.

Reply:
Thanks @bharatviswa504 
I found the check also can be done by the base class method,
so I fix it and add some tests.